enhancements/sig-policy/89-operator-policy-kind/README.md

```diff
@@ -0,0 +1,310 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+  [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+Enhance the policy framework with a new kind, `OperatorPolicy`, to allow for better management of
+OLM Operator installations on managed clusters. The controller for this kind would be included in
+the configuration-policy-controller container, on managed clusters. The new controller would better
+handle imperative installation steps for OLM objects, and provide better support for upgrade and
+uninstallation scenarios.
+
+## Motivation
+
+It is currently possible to install OLM Operators on managed clusters with `ConfigurationPolicies`,
+but it can be complex and error-prone. For example, multiple policies are often needed in order to
+manage the different OLM resources such as OperatorGroup, Subscription, InstallPlan, and
+ClusterServiceVersion. Upgrading or uninstalling an operator can also be difficult with
+ConfigurationPolicies, since it is not generally as easy as just updating one value or deleting a
+resource.
+
+Creating a new policy type will help streamline the process of declaring an operator to be installed
+on a cluster. A new controller will help cover specialized cases that come up when managing operator
+installations, rather than teaching the existing configuration-policy-controller to deal with
+specific types.
+
+### Goals
+
+1. It should be easy to install an operator on a managed cluster with just one policy.
+2. It should be possible to limit updates to operators on clusters, so they do not upgrade beyond a
+   known good version defined in the policy.
+3. The policy managing the operator should clearly report back the status of the installation,
+   including the operator pod's health, with clear violation messages if something is not right.
+4. What happens when the policy is deleted should be configurable: the operator might be deleted, or
+   the operator and its CRDs might be deleted, or nothing might be deleted.
+
+### Non-Goals
+
+1. The installation of OLM on managed clusters will not be automated by this enhancement.
+2. Management of "Operands" will not be included in this new type - they should be defined in
+   separate `ConfigurationPolicies`, possibly with a
+   [dependency](../74-order-policy-execution/README.md) on the `OperatorPolicy`.
+3. Management of `CatalogSources` will not be included in the new type - they should be defined in
+   separate `ConfigurationPolicies`.
+
+## Proposal
+
+### OperatorPolicy Syntax
+
+```yaml
+apiVersion: policy.open-cluster-management.io/v1beta1
+kind: OperatorPolicy
+metadata:
+  name: my-operator
+spec:
+  remediationAction: enforce # or inform
```

Comment:
What is checked in `inform` mode? What kind of non-compliant message would you get?